Fix part of #8668: JSOC added for admin-router.services

anumehaagrawal opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #8668 
2. This PR does the following: Adds documentation for functions in admin-router.services.ts

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

anumehaagrawal committed:
JSDoc for Admin router services added

anumehaagrawal committed:
Fix JSDoc admin-router

anumehaagrawal committed:
Lint Fix JSDoc admin-router

anumehaagrawal commented:
@nithusha21 Could you add a PR_CHANGELOG label for this?

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -25,11 +25,18 @@ import { AdminPageConstants } from
     AdminPageConstants.ADMIN_TAB_URLS.ACTIVITIES);
 
+  /**
+   * @param {string} tabHash tab url hash value.
```
For all docstring descriptions: please start with a capital letter.

Also, try to make them clearly readable. If you just say "tab url hash value", I don't think that developers will clearly understand it. Use full English sentences/clauses instead, don't abbreviate things or leave out words. Try to explain it to someone who doesn't know the code.

anumehaagrawal replied:
Sure ill change it

seanlip replied:
Thanks -- just a tip, though: please make the changes first, and then submit the comments to these at the same time as you submit the code. 

In general, please make sure to follow the instructions [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change). Otherwise, it makes it harder for reviewers to keep track of these comments.

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -25,11 +25,18 @@ import { AdminPageConstants } from
+   * @returns {string|null} tab name of the hash value. If no match then null.
+   */
   getTabNameByHash(tabHash: string): string | null {
+    // Iterates through the ADMIN_TAB_URLS map and returns the
```
Descriptions can go above the `@param` etc. parts of the docstring. See https://jsdoc.app/about-getting-started.html for examples.

anumehaagrawal replied:
Will change it

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -40,45 +47,46 @@ export class AdminRouterService {
+   * @param {string} tabHash tab url hash value.
    */
   showTab(tabHash: string): void {
+    // Navigates the page to the specified tab based on its HTML hash.
```
Ditto.

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -40,45 +47,46 @@ export class AdminRouterService {
 
   /**
-   * Returns whether the activities tab is open.
+   * @returns {boolean} returns whether the activities tab is open.
```
`@returns {boolean} Whether the activities tab is open.`

Ditto below.

seanlip reviewed (changes requested):
Thanks @anumehaagrawal! I took a first pass.

nithusha21 commented:
Thanks for reviewing @seanlip! I don't have anything to add!

anumehaagrawal committed:
Comment fixes on PR

anumehaagrawal committed:
Lint errors fixed

anumehaagrawal commented:
![image](https://user-images.githubusercontent.com/25412153/94385352-ce201d80-00f9-11eb-9772-1484d08dbf9d.png)
I have been trying to push my changes for a while now and I keep running into errors because of files that I havent modified. In this as well, i havent touched these files. Should i make the changes in the common file as shown for linting errors to pass?
@seanlip @DubeySandeep

seanlip commented:
Hi @anumehaagrawal -- that's a bit odd. I'm not sure why you're running into that issue.

Could you merge to get your branch up to date with develop (see step 6 [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change)) and then try pushing again? Typically the linter should only check the list of files that are being changed in the PR. If it's checking more stuff then it likely means that you're out-of-sync with HEAD of develop.

If you're still running into problems, let us know.

seanlip commented:
/cc @Hudda

anumehaagrawal committed:
Merge remote-tracking branch 'upstream/develop' into jsdoc-admin-router

anumehaagrawal commented:
@seanlip Yep I did that and each time I try to push I run into a new lint error from other files. Not sure if this is happening with others also.

anumehaagrawal commented:
@seanlip I have pushed the changes from review comments PTAL!

seanlip commented:
@anumehaagrawal Please address all review comments (following the "instructions for making a PR" on the getting-started wiki page), and only request a review after that. Thanks!

DubeySandeep commented:
> ![image](https://user-images.githubusercontent.com/25412153/94385352-ce201d80-00f9-11eb-9772-1484d08dbf9d.png)
> I have been trying to push my changes for a while now and I keep running into errors because of files that I havent modified. In this as well, i havent touched these files. Should i make the changes in the common file as shown for linting errors to pass?
> @seanlip @DubeySandeep

IIRC, @savithakj also faced a similar issue recently with some untouched/unchanged files (The same import order issue). Deleting the existing oppia_tools dir and ran the git-push command, this helped her resolve the issue. Can you please try this? [@Hudda Though we need to see what's going and make sure that it should not block developers!]

DubeySandeep commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -25,10 +25,18 @@ import { AdminPageConstants } from
 @Injectable({
   providedIn: 'root'
 })
+
```
There shouldn't be any newline here, right? (In between decorator and class definition)

anumehaagrawal replied:
Removed the space

DubeySandeep commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -49,36 +58,36 @@ export class AdminRouterService {
 
   /**
-   * Returns whether the activities tab is open.
+   * @returns {boolean} Returns whether the activities tab is open.
```
```suggestion
   * @returns {boolean} Whether the activities tab is open.
```
The description should be about the returned value.

DubeySandeep commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -25,10 +25,18 @@ import { AdminPageConstants } from
+   * Iterates through the ADMIN_TAB_URLS map and returns the
+   * tab name corresponding to the hash.
+   * @param {string} tabHash Represents hash name of a tab URL.
+   * @returns {string|null} Tab name of the hash value.
+   * If no match then null.
```
```suggestion
   * @returns {string|null} Tab name corresponding to the given the hash value 
   *     if found else null.
```

DubeySandeep commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -49,36 +58,36 @@ export class AdminRouterService {
 
   /**
-   * Returns whether the jobs tab is open.
+   * @returns {boolean} Returns whether the jobs tab is open.
```
```suggestion
   * @returns {boolean} Whether the jobs tab is open.
```
Ditto elsewhere

DubeySandeep reviewed (changes requested):
@anumehaagrawal, Thanks for the PR, I've left a few minor comments PTAL!

anumehaagrawal committed:
Update core/templates/pages/admin-page/services/admin-router.service.ts

Co-authored-by: Sandeep Dubey <dubeysandeep.in@gmail.com>

anumehaagrawal committed:
Update core/templates/pages/admin-page/services/admin-router.service.ts

Co-authored-by: Sandeep Dubey <dubeysandeep.in@gmail.com>

anumehaagrawal committed:
Update core/templates/pages/admin-page/services/admin-router.service.ts

Co-authored-by: Sandeep Dubey <dubeysandeep.in@gmail.com>

anumehaagrawal committed:
Fixed return comments

anumehaagrawal committed:
Merge remote-tracking branch 'upstream/develop' into jsdoc-admin-router

anumehaagrawal committed:
Adding fetched request

anumehaagrawal commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -49,36 +57,36 @@ export class AdminRouterService {
 
   /**
-   * Returns whether the activities tab is open.
+   * @returns {boolean} Whether the activities tab is open.
```
Done!

anumehaagrawal commented:
> @anumehaagrawal Please address all review comments (following the "instructions for making a PR" on the getting-started wiki page), and only request a review after that. Thanks!

I have addressed all the review comments

anumehaagrawal commented:
@DubeySandeep PTAL!

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -29,6 +29,13 @@ export class AdminRouterService {
+  /**
+   * Iterates through the ADMIN_TAB_URLS map and returns the
+   * tab name corresponding to the hash.
+   * @param {string} tabHash Represents hash name of a tab URL.
```
`tabHash The string after the '#' character in the URL.`

anumehaagrawal replied:
Done

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -29,6 +29,13 @@ export class AdminRouterService {
+   * Iterates through the ADMIN_TAB_URLS map and returns the
+   * tab name corresponding to the hash.
+   * @param {string} tabHash Represents hash name of a tab URL.
+   * @returns {string|null} Tab name of the hash value
```
`The corresponding tab name, or null if the URL hash does not correspond to a valid tab.`

(Also, indent the continuation line by 2.)

anumehaagrawal replied:
Done

seanlip commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -41,6 +48,7 @@ export class AdminRouterService {
 
   /**
    * Navigates the page to the specified tab based on its HTML hash.
+   * @param {string} tabHash Represents hash name of a tab URL.
```
`tabHash The string after the '#' character in the URL.`

anumehaagrawal replied:
Done

seanlip reviewed (changes requested):
Thanks @anumehaagrawal! Left a few suggestions, but other than that it should be good to go.

@nishantwrp -- just to check, are we still putting types in the docstrings, or are those redundant due to the typescript stuff? (We should update the original starter issue with the decision.)

nishantwrp commented:
@seanlip I agree we can skip writing types in jsdoc because we use typescript.

DubeySandeep reviewed (approved):
Approving this PR as @seanlip has covered all the files and I don't have any other comments for the PR.

anumehaagrawal committed:
Fixed ciomments after review

anumehaagrawal committed:
Merge remote-tracking branch 'upstream/develop' into jsdoc-admin-router

anumehaagrawal commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -41,6 +48,7 @@ export class AdminRouterService {
 
   /**
    * Navigates the page to the specified tab based on its HTML hash.
+   * @param {string} tabHash The string after the '#' character in the URL.
```
Changed the text

anumehaagrawal commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -29,6 +29,13 @@ export class AdminRouterService {
+   * Iterates through the ADMIN_TAB_URLS map and returns the
+   * tab name corresponding to the hash.
+   * @param {string} tabHash The string after the '#' character in the URL.
+   * @returns {string|null} The corresponding tab name, or null
```
Changed as needed

seanlip replied:
Hi @anumehaagrawal -- just a note, you don't need to write this. Replying "Done" to the original conversation thread is enough. That helps keep all the conversation in a single place.

Thanks!

anumehaagrawal commented on core/templates/pages/admin-page/services/admin-router.service.ts:
```diff
@@ -29,6 +29,13 @@ export class AdminRouterService {
+   * Iterates through the ADMIN_TAB_URLS map and returns the
+   * tab name corresponding to the hash.
+   * @param {string} tabHash The string after the '#' character in the URL.
+   * @returns {string|null} The corresponding tab name, or null
```
Changed as needed

seanlip reviewed (approved):
LGTM. Thanks @anumehaagrawal!

@nithusha21 could you PTAL as codeowner?
